Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/self_intersections.h

```diff
@@ -238,6 +238,9 @@ self_intersections_impl(const FaceRange& face_range,
   // This loop is very cheap, so there is hardly anything to gain from parallelizing it
   for(face_descriptor f : face_range)
   {
+    if (f == boost::graph_traits<TriangleMesh>::null_face())
```

Comment:
how can this be possible? are you calling the function from a range of halfedge, some of them being on the border?

Reply:
this commit was part of the PR #4192
I am pretty sure I had met this case in the test case I was running at that time...
I cost of that check is low and prevents for misusing the function, so I think we can keep it

Reply:
@MaelRL what do you think? I'm not in favor of adding it and would prefer fixing the calling code.

Reply:
I also would prefer if this function assumed a clean range.

Reply:
On the other hand, being robust to users' errors is also a nice property, no?

Reply:
Well, I usually agree but here we are talking about a NULL element in an input range.

Reply:
So the actual bug is in the calling code, from `demo/Polyhedron/`, right?